src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
+
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
+}
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export function isTriangleOk(a: number, b: number, c: number): boolean {
+  const sides = [a, b, c];
+
+  sides.sort((sideA: number, sideB: number) => sideB - sideA);
 
+  return sides[0] < sides[1] + sides[2];
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || !isTriangleOk(a, b, c)) {
```

Comment:
Create a helper function that accepts any number of args and checks them to be > 0. Reuse it with all shapes.